testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/mask.tests.js

```diff
@@ -2093,15 +2093,19 @@ QUnit.module('stubs', moduleConfig, function() {
         { format: '#,##0\'.\'\'0\'\'0\'', expectedText: '1,234.00', expectedValue: 1234 },
         { format: '\'-\'#,###.00', expectedText: '-1.00', expectedValue: 1 },
         { format: '\'-\'#,###.00', expectedText: '--1.00', expectedValue: -1 },
-    ].forEach(({ format, expectedText, expectedValue }) => {
+        { format: '#', expectedText: '1', expectedValue: 1 },
+        { format: '#', expectedText: '-1', expectedValue: -1 },
+        { format: '#', expectedText: '', typedText: '-', expectedValue: null },
+        { format: '#', expectedText: '-1', typedText: '1-', expectedValue: -1 },
+    ].forEach(({ format, expectedText, typedText, expectedValue }) => {
         QUnit.test(`widget should correctly apply format="${format}", value="${expectedValue}"`, function(assert) {
             this.instance.option({
                 value: null,
                 format
             });
 
             this.keyboard
-                .type(expectedValue.toString())
+                .type(typedText || expectedValue.toString())
```

Comment:
```suggestion
                .type(typedText ?? expectedValue.toString())
```

Reply:
?? doesn't supported in our test environment